### PR TITLE
Pass control events to nodes for emitting to subscribers

### PR DESCRIPTION
**Pull request by OverloadUT**
This adds quick and dirty support for emitting "control" events from the ISY event stream out to subscribers via the nodes associated with the control events.

The primary use case I am trying to solve with this is allowing Home Assistant to be aware of button presses on switches, not just when a light state changes. This will allow creating special automations for when buttons are double-pressed, or even using an Insteon controller (such as the 8-button remote) without it needing to be linked to any actual devices in the Insteon network.

It will ALSO open the door to properly handling heartbeats, which are always "on" events. Since they never emit an "off", the events never make it out to Home Assistant.

Implements #33

### CODE REVIEW

File: PyISY/Events/events.py [hunk 1]
```diff
@@ -62,22 +61,27 @@ def _routemsg(self, msg):
...
+                self.parent.nodes._controlmsg(xmldoc)
+            elif cntrl == '_11':  # WEATHER UPDATE
+                if self.parent.configuration['Weather Information']:
+                    self.parent.climate._upmsg(xmldoc)
+            elif cntrl == '_1':  # VARIABLE OR PROGRAM UPDATE
+                if '<var' in msg:  # VARIABLE
+                    self.parent.variables._upmsg(xmldoc)
+                elif '<id>' in msg:  # PROGRAM
+                    self.parent.programs._upmsg(xmldoc)
+        except:
```
Comment (rmkraus) [inline PyISY/Events/events.py]: This feels awfully dirty. Why the naked except statement here?

File: PyISY/Events/eventsSSL.py [hunk 2]
```diff
@@ -76,25 +75,30 @@ def _routemsg(self, msg):
...
+                    self.parent.climate._upmsg(xmldoc)
+            elif cntrl == '_1':  # VARIABLE OR PROGRAM UPDATE
+                if '<var' in msg:  # VARIABLE
+                    self.parent.variables._upmsg(xmldoc)
+                elif '<id>' in msg:  # PROGRAM
+                    self.parent.programs._upmsg(xmldoc)
+                else:  # SOMETHING HAPPENED WITH A PROGRAM FOLDER
+                    # but they ISY didn't tell us what, so...
+                    self.parent.programs.update()
+        except:
```
Comment (rmkraus) [inline PyISY/Events/eventsSSL.py]: Same question as before.

File: PyISY/Nodes/__init__.py [hunk 3]
```diff
@@ -141,6 +141,20 @@ def _upmsg(self, xmldoc):
...
 
+    def _controlmsg(self, xmldoc):
+        """Passes Control events from an event stream message to nodes, for
+        sending out to subscribers."""
+        try:
+            nid = xmldoc.getElementsByTagName('node')[0].firstChild.toxml()
+            cntrl = xmldoc.getElementsByTagName('control')[0].firstChild.toxml()
+            self.getByID(nid).controlEvents.notify(cntrl)
+            self.parent.log.info('ISY Node Control Event: ' + nid + ' ' + cntrl)
+        except:
```
Comment (rmkraus) [inline PyISY/Nodes/__init__.py]: I understand what you're doing here, but is there a better way to ensure the XML contains the necessary elements without using try/except? Possibly:

```python
nodes = xmldoc.getElementsByTagName('node')
ctrls = xmldoc.getElementsByTagName('control')

if len(nodes) > 0 and len(cntrls) > 0:
    nid = nodes[0].firstChild.toxml()
    cntrl = cntrls[0].firstChild.toxml()
    self.getByID(nid).controlEvents.notify(cntrl)
    self.parent.log.info('ISY Node Control Event: ' + nid + ' ' + cntrl)
```

File: PyISY/Events/events.py (hunk 1 above)
Comment (OverloadUT) [reply PyISY/Events/events.py]: This... must have been an oversight. I don't remember wrapping that entire block in a naked except, but yeah, it's got to go. I'll fix it.

File: PyISY/Nodes/__init__.py (hunk 3 above)
Comment (OverloadUT) [reply PyISY/Nodes/__init__.py]: It most definitely shouldn't be a catchall `except`, but when looking around it seemed to be a common belief that the Pythonic way to handle checking if something exists is to just try it and catch it, which is why I went with this pattern. I can change it to however you'd like it to be though

Comment (rmkraus) [reply PyISY/Nodes/__init__.py]: Meh, I'm kind of 50-50 on this one. You can keep it how it is, but a more specific except clause would be good.

**Comment by OverloadUT** [comment]
Latest commits remove all of the wildcard exception clauses! Also fixed one of the ones that was already present in the library.

**Comment by rmkraus** [comment]
👍

**Pull request merged**